### PR TITLE
feat: nibbles

### PR DESCRIPTION
**Pull request by justcode740**

### CODE REVIEW

File: plonky2x/src/frontend/vars/byte.rs [hunk 1]
```diff
@@ -299,11 +310,12 @@ mod tests {
         let mut builder = CircuitBuilder::<F, D>::new();
 
         let value = rand::random::<u8>();
-        let byte = builder.constant(value);
-        let nibbles = byte.to_nibbles(&mut builder);
+        let byte = builder.constant::<ByteVariable>(value);
+        let _nibbles = byte.to_nibbles(&mut builder);
 
         // TODO: test that the nibbles are correct
```
Comment (puma314) [inline plonky2x/src/frontend/vars/byte.rs:316]: This is awesome! Let's just add a test here, I can show you how to do that!

Comment (justcode740) [reply plonky2x/src/frontend/vars/byte.rs:316]: oh, just saw the this, i added a weird test...

File: plonky2x/src/frontend/vars/byte.rs [hunk 2]
```diff
@@ -99,7 +99,18 @@ impl ByteVariable {
         self,
         builder: &mut CircuitBuilder<F, D>,
     ) -> [ByteVariable; 2] {
-        todo!()
+        // this function split the bytevariables [abcdefgh] to [0000abcd] and [0000efgh], please implement it 
+        // let bits = self.to_be_bits();
```
Comment (puma314) [inline plonky2x/src/frontend/vars/byte.rs:103]: nit: can we get rid of this comment? otherwise looks good to me!

**Review by puma314 — APPROVED** [review]
Small nit, otherwise looks good to me to merge!

**Pull request merged**